development-docs/systemtests/io.strimzi.systemtest.cruisecontrol.CruiseControlST.md

```diff
@@ -0,0 +1,218 @@
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Initialize JBOD storage configuration | JBOD storage with specific disk sizes are initialized |
+| 2. | Create Kafka broker and controller pools using the initialized storage | Kafka broker and controller pools are created and available |
+| 3. | Deploy Kafka with Cruise Control enabled | Kafka deployment with Cruise Control is successfully created |
+| 4. | Create Kafka Rebalance resource with disk rebalancing configured | Kafka Rebalance resource is created and configured for disk balancing |
+| 5. | Wait for the Kafka Rebalance to reach the 'ProposalReady' state | Kafka Rebalance resource reaches the 'ProposalReady' state |
+| 6. | Check the status of the Kafka Rebalance for intra-broker disk balancing | The 'provisionStatus' in the optimization result is 'UNDECIDED' |
+
+**Labels:**
+
+* [cruise-control](labels/cruise-control.md)
+
+
+## testCruiseControlReplicaMovementStrategy
+
+**Description:** Test that verifies the configuration and application of custom Cruise Control replica movement strategies.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Create broker and controller KafkaNodePools. | Both KafkaNodePools are successfully created |
+| 2. | Create Kafka and Cruise Control resources | Kafka and Cruise Control resources are created and deployed |
+| 3. | Verify default Cruise Control replica movement strategy | Default replica movement strategy is verified in the configuration |
+| 4. | Update Cruise Control configuration with non-default replica movement strategies | Cruise Control configuration is updated with new strategies |
+| 5. | Ensure that Cruise Control pod is rolled due to configuration change | Cruise Control pod is rolled and new configuration is applied |
+| 6. | Verify the updated Cruise Control configuration | Updated replica movement strategies are verified in Cruise Control configuration |
+
+**Labels:**
+
+* [cruise-control](labels/cruise-control.md)
+
+
+## testCruiseControlTopicExclusion
+
+**Description:** Verify that Kafka Cruise Control excludes specified topics and includes others.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Set topic names | Topic names are set |
+| 2. | Create broker and controller KafkaNodePools. | Both KafkaNodePools are successfully created |
+| 3. | Deploy Kafka with Cruise Control enabled | Kafka cluster with Cruise Control is deployed |
+| 4. | Create topics to be excluded and included | Topics 'excluded-topic-1', 'excluded-topic-2', and 'included-topic' are created |
+| 5. | Create KafkaRebalance resource excluding specific topics | KafkaRebalance resource is created with 'excluded-.*' topics pattern |
+| 6. | Wait for KafkaRebalance to reach ProposalReady state | KafkaRebalance reaches the ProposalReady state |
+| 7. | Check optimization result for excluded and included topics | Excluded topics are in the optimization result and included topic is not |
```

Comment:
Is that right? Excluded in the optimization result and included not?

Reply:
Yes that's correct
```
 KafkaRebalanceStatus kafkaRebalanceStatus = KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getClusterName()).get().getStatus();
        assertThat(kafkaRebalanceStatus.getOptimizationResult().get("excludedTopics").toString(), containsString(excludedTopic1));
        assertThat(kafkaRebalanceStatus.getOptimizationResult().get("excludedTopics").toString(), containsString(excludedTopic2));
        assertThat(kafkaRebalanceStatus.getOptimizationResult().get("excludedTopics").toString(), not(containsString(includedTopic)));
```